src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/UserService.kt

```diff
@@ -284,6 +292,9 @@ class UserService(
       }
     }
 
+    val staffProbationDeliveryUnit =
+      staffUserDetails.teams?.firstOrNull()?.borough?.let { probationDeliveryUnitRepository.findByDeliusCode(it.code) }
```

Comment:
i appreciate you're following an established pattern but i think it may be good to introduce a private function that returns a nullable ProbationDeliveryUnitEntity given StaffUserDetails and then use that in the create/update methods to avoid code duplication